src/Microsoft.ML.Data/Utils/ApiUtils.cs

```diff
@@ -23,7 +23,8 @@ private static OpCode GetAssignmentOpCode(Type t)
             if (t == typeof(ReadOnlyMemory<char>) || t == typeof(string) || t.IsArray ||
                 (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(VBuffer<>)) ||
                 (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Nullable<>)) ||
-                t == typeof(DateTime) || t == typeof(DateTimeOffset) || t == typeof(TimeSpan) || t == typeof(DataViewRowId))
+                t == typeof(DateTime) || t == typeof(DateTimeOffset) || t == typeof(TimeSpan) ||
+                t == typeof(DataViewRowId) || TypeManager.GetDataViewTypeOrNull(t) != null)
```

Comment:
>GetDataViewTypeOrNull [](start = 58, length = 21)

let's just add a Contains() method instead of checking for null.

Reply:
Ok but will call it `Knows(...)`.

---
In reply to: [285269922](https://github.com/dotnet/machinelearning/pull/3263#discussion_r285269922) [](ancestors = 285269922)